src/js.c

```diff
@@ -1911,14 +1913,14 @@ _fetch(natsMsgList *list, natsSubscription *sub, jsFetchRequest *req, bool simpl
         // If we come from natsSubscription_Fetch() (simpleFetch is true), then
         // we decide on the NoWait value.
         if (simpleFetch)
-            noWait = (req->Batch - count > 1 ? true : false);
+            noWait = (batch - count > 1 ? true : false);
         else
             noWait = req->NoWait;
     }
 
     // If we have OK and not all messages, we will send a fetch
     // request to the server.
-    while ((s == NATS_OK) && (count != req->Batch) && ((req->MaxBytes == 0) || (size < req->MaxBytes)))
+    while ((s == NATS_OK) && (count < batch) && ((req->MaxBytes == 0) || (size < req->MaxBytes)))
```

Comment:
I think that this is just before this `while` loop that we would need to use/define `batch = req->Batch`. Everything that is above is not changing `req->Batch`. And if we use `batch` here as the value of `req->Batch` when entering the loop and `batch` does not change (but `req->Batch` may), then the `!=` would still be correct, there would not be a need to change to `<`.

Reply:
Fixed